File: How to write an SFTP file transfer client for WinForms/[C#]-How to write an SFTP file transfer client for WinForms/C#/LoginInfo.cs
using System;
using System.Diagnostics;
using ComponentPro.IO;
using ComponentPro.Net;

namespace ClientSample
{
    public class LoginInfo
    {
        public const string LogLevel = "LogLevel";
        public const string ServerName = "ServerName";
        public const string ServerPort = "ServerPort";
        public const string UserName = "Username";
        public const string Password = "Password";
        public const string RemoteDir = "RemoteDir";
        public const string LocalDir = "LocalDir";

        public const string Utf8Encoding = "Utf8Encoding";

        #region Proxy

        public const string ProxyServer = "ProxyServer";
        public const string ProxyPort = "ProxyPort";
        public const string ProxyUser = "ProxyUser";
        public const string ProxyPassword = "ProxyPassword";
        public const string ProxyDomain = "ProxyDomain";
        public const string ProxyType = "ProxyType";
        public const string ProxyHttpAuthnMethod = "ProxyHttpAuthnMethod";

        #endregion

        public const string SyncMethod = "SyncMethod";
        public const string SyncResumability = "SyncResumability";
        public const string SyncDateTime = "SyncDateTime";
        public const string SyncRecursive = "SyncRecursive";
        public const string SyncSearchPattern = "SyncSearchPattern";

        public const string ServerType = "ServerType";

        #region Methods

        public static void SaveConfig(SettingInfoBase settings)
        {
            // Save Login information.
            settings.SaveProperty(ServerName);
            settings.SaveProperty(ServerPort);
            settings.SaveProperty(UserName);
            settings.SaveProperty(Password);
            settings.SaveProperty(RemoteDir);
            settings.SaveProperty(LocalDir);

            settings.SaveProperty(Utf8Encoding);
            
            // Proxy Info.
            settings.SaveProperty(ProxyServer);
            settings.SaveProperty(ProxyPort);
            settings.SaveProperty(ProxyUser);
            settings.SaveProperty(ProxyPassword);
            settings.SaveProperty(ProxyDomain);
            settings.SaveProperty(ProxyType);
            settings.SaveProperty(ProxyHttpAuthnMethod);

            settings.SaveProperty(SyncMethod);
            settings.SaveProperty(SyncResumability);
            settings.SaveProperty(SyncDateTime);
            settings.SaveProperty(SyncRecursive);
            settings.SaveProperty(SyncSearchPattern);

            settings.SaveProperty(LogLevel);

            settings.SaveProperty(ServerType);
        }

        public static void LoadConfig(SettingInfoBase settings)
        {
            // Server and authentication info.
            settings.LoadProperty(ServerName, string.Empty);
            settings.LoadProperty(ServerPort, 21);
            settings.LoadProperty(UserName, string.Empty);
            settings.LoadProperty(Password, string.Empty);
            settings.LoadProperty(RemoteDir, string.Empty);
            settings.LoadProperty(LocalDir, AppDomain.CurrentDomain.BaseDirectory);
            
            settings.LoadProperty(Utf8Encoding, false);

            // Proxy info.
            settings.LoadProperty(ProxyServer, string.Empty);
            settings.LoadProperty(ProxyPort, 1080);
            settings.LoadProperty(ProxyUser, string.Empty);
            settings.LoadProperty(ProxyPassword, string.Empty);
            settings.LoadProperty(ProxyDomain, string.Empty);
            settings.LoadProperty(ProxyType, 0);
            settings.LoadProperty(ProxyHttpAuthnMethod, 0);

            settings.LoadProperty(SyncMethod, 0);
            settings.LoadProperty(SyncResumability, false);
            settings.LoadProperty(SyncDateTime, true);
            settings.LoadProperty(SyncRecursive, (int)RecursionMode.RecurseIntoAllSubFolders);
            settings.LoadProperty(SyncSearchPattern, "*.*");

            settings.LoadProperty(LogLevel, (int)TraceEventType.Information);

            settings.LoadProperty(ServerType, 0);
        }

        #endregion
    }    
}

File: FTP Client for WinForms using Ultimate FTP library/[VB]-FTP Client for WinForms using Ultimate FTP library/VB/LoginInfo.vb
Imports ComponentPro.IO
Imports ComponentPro.Net

Namespace ClientSample
	Public Class LoginInfo
		Public Const LogLevel As String = "LogLevel"
		Public Const ServerName As String = "ServerName"
		Public Const ServerPort As String = "ServerPort"
		Public Const UserName As String = "Username"
		Public Const Password As String = "Password"
		Public Const RemoteDir As String = "RemoteDir"
		Public Const LocalDir As String = "LocalDir"

		Public Const Utf8Encoding As String = "Utf8Encoding"

		#Region "Proxy"

		Public Const ProxyServer As String = "ProxyServer"
		Public Const ProxyPort As String = "ProxyPort"
		Public Const ProxyUser As String = "ProxyUser"
		Public Const ProxyPassword As String = "ProxyPassword"
		Public Const ProxyDomain As String = "ProxyDomain"
		Public Const ProxyType As String = "ProxyType"
		Public Const ProxyHttpAuthnMethod As String = "ProxyHttpAuthnMethod"

		#End Region

		Public Const SyncMethod As String = "SyncMethod"
		Public Const SyncResumability As String = "SyncResumability"
		Public Const SyncDateTime As String = "SyncDateTime"
		Public Const SyncRecursive As String = "SyncRecursive"
		Public Const SyncSearchPattern As String = "SyncSearchPattern"

		#Region "Methods"

		Public Shared Sub SaveConfig(ByVal settings As SettingInfoBase)
			' Save Login information.
			settings.SaveProperty(ServerName)
			settings.SaveProperty(ServerPort)
			settings.SaveProperty(UserName)
			settings.SaveProperty(Password)
			settings.SaveProperty(RemoteDir)
			settings.SaveProperty(LocalDir)

			settings.SaveProperty(Utf8Encoding)

			' Proxy Info.
			settings.SaveProperty(ProxyServer)
			settings.SaveProperty(ProxyPort)
			settings.SaveProperty(ProxyUser)
			settings.SaveProperty(ProxyPassword)
			settings.SaveProperty(ProxyDomain)
			settings.SaveProperty(ProxyType)
			settings.SaveProperty(ProxyHttpAuthnMethod)

			settings.SaveProperty(SyncMethod)
			settings.SaveProperty(SyncResumability)
			settings.SaveProperty(SyncDateTime)
			settings.SaveProperty(SyncRecursive)
			settings.SaveProperty(SyncSearchPattern)

			settings.SaveProperty(LogLevel)
		End Sub

		Public Shared Sub LoadConfig(ByVal settings As SettingInfoBase)
			' Server and authentication info.
			settings.LoadProperty(ServerName, String.Empty)
			settings.LoadProperty(ServerPort, 21)
			settings.LoadProperty(UserName, String.Empty)
			settings.LoadProperty(Password, String.Empty)
			settings.LoadProperty(RemoteDir, String.Empty)
			settings.LoadProperty(LocalDir, AppDomain.CurrentDomain.BaseDirectory)

			settings.LoadProperty(Utf8Encoding, False)

			' Proxy info.
			settings.LoadProperty(ProxyServer, String.Empty)
			settings.LoadProperty(ProxyPort, 1080)
			settings.LoadProperty(ProxyUser, String.Empty)
			settings.LoadProperty(ProxyPassword, String.Empty)
			settings.LoadProperty(ProxyDomain, String.Empty)
			settings.LoadProperty(ProxyType, 0)
			settings.LoadProperty(ProxyHttpAuthnMethod, 0)

			settings.LoadProperty(SyncMethod, 0)
			settings.LoadProperty(SyncResumability, False)
			settings.LoadProperty(SyncDateTime, True)
			settings.LoadProperty(SyncRecursive, CInt(RecursionMode.RecurseIntoAllSubFolders))
			settings.LoadProperty(SyncSearchPattern, "*.*")

			settings.LoadProperty(LogLevel, CInt(TraceEventType.Information))
		End Sub

		#End Region
	End Class
End Namespace